UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs

```diff
@@ -186,10 +186,17 @@ private void ProcessWindNodes(MetaDataNode windyNode)
 
 				correctedForce = Mathf.Clamp(correctedForce, 0, 30);
 
-				pushable.NewtonianPush(registerTile.transform.rotation * (Vector2)windyNode.WindDirection, Random.Range((float)(correctedForce * 0.8), correctedForce),  spinFactor: Random.Range(1, 150));
+				if (pushable.CanBeWindPushed)
+				{
+					pushable.NewtonianPush(registerTile.transform.rotation * (Vector2)windyNode.WindDirection, Random.Range((float)(correctedForce * 0.8), correctedForce),  spinFactor: Random.Range(1, 150));
+				}
+
+
 				if (pushable.stickyMovement && windyNode.WindForce > 3)
 				{
-					if (pushable is MovementSynchronisation && windyNode.WindForce < (int)WindStrength.STRONG) continue;
+					var Playermove = pushable as MovementSynchronisation;
+					if ( windyNode.WindForce < (int)WindStrength.STRONG && Playermove == null) continue;
+					if (Playermove.CanBeWindPushed == false) continue;
```

Comment:
```suggestion
					var playerMove = pushable as MovementSynchronisation;
					if ( windyNode.WindForce < (int)WindStrength.STRONG && playerMove == null) continue;
					if (playerMove .CanBeWindPushed == false) continue;
```

Reply:
The ```&& playerMove == null``` is that in the right place?

Reply:
null check is more expensive than magnitude

Reply:
yeah, not asking about sequence. You are checking playerMove after that If and you are not returning if it is null before. Is that supposed to be an ```||```?

Reply:
well the as operation just casts, and then you have to check it to make sure it is not null